endpoint/endpoint.go

```diff
@@ -293,12 +293,20 @@ func (e *Endpoint) String() string {
 // only endpoints that match.
 func FilterEndpointsByOwnerID(ownerID string, eps []*Endpoint) []*Endpoint {
 	filtered := []*Endpoint{}
+	visited := make(map[EndpointKey]bool) // Initialize the visited map
 	for _, ep := range eps {
+		key := EndpointKey{DNSName: ep.DNSName, RecordType: ep.RecordType, SetIdentifier: ep.SetIdentifier}
```

Comment:
What are the possible cases in which we have duplicated endpoints in this loop? From my understanding, it's only for overlapping zones, is this right?